databuilder/databuilder/extractor/snowflake_usage_extractor.py

```diff
@@ -0,0 +1,106 @@
+"""
+The Snowflake Usage Extractor allows you to populate the "Frequent Users" and "Popular Tables" features within
+Amundsen with the help of the TableColumnUsage class. The way you calculate popularity for Snowflake is custom
+and an example of how to calculate table popularity can be found in the sample_snowflake_table_usage.scala script.
+"""
+
+import logging
+from typing import (
+    Any, Dict, Iterator, Union,
+)
+
+from pyhocon import ConfigFactory, ConfigTree
+
+from databuilder import Scoped
+from databuilder.extractor import sql_alchemy_extractor
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.sql_alchemy_extractor import SQLAlchemyExtractor
+from databuilder.models.table_column_usage import ColumnReader, TableColumnUsage
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SnowflakeUsageExtractor(Extractor):
+    # SELECT statement from snowflake table that contains usage data
+    SQL_STATEMENT = """
+        SELECT
+            database,
+            schema,
+            name,
+            user_email,
+            read_count
+        FROM
+            {database}.{schema}.{table}
+        {where_clause_suffix};
+    """
+    # CONFIG KEYS
+    WHERE_CLAUSE_SUFFIX_KEY = 'where_clause_suffix'
+    POPULARITY_TABLE_DATABASE = 'popularity_table_database'
+    POPULARTIY_TABLE_SCHEMA = 'popularity_table_schema'
+    POPULARITY_TABLE_NAME = 'popularity_table_name'
+
+    DEFAULT_CONFIG = ConfigFactory.from_dict(
+        {WHERE_CLAUSE_SUFFIX_KEY: ' ',
```

Comment:
a side-note: sql query could as well have hardcoded `WHERE` close and default condition would then be 1=1. With current approach users also need to provide `WHERE` keyword here, am I right?

Reply:
yes users have to provide the `WHERE` keyword. I used the same convention as the `snowflake_metadata_extractor`: https://github.com/amundsen-io/amundsen/blob/main/databuilder/databuilder/extractor/snowflake_metadata_extractor.py#L70